src/heliaServer.ts

```diff
@@ -88,7 +81,17 @@ export class HeliaServer {
     }
     const finalUrl = `//${cidv1Address ?? address}.${namespace}.${request.hostname}${relativePath}`
     this.log('Redirecting to final URL:', finalUrl)
-    await reply.redirect(finalUrl)
+    // in an ideal world we would just redirect to the final URL, but to make this compatible with
+    // tiros, just send the content with a 301. It's also faster
+    return this.fetch({
+      request: {
+        ...request,
+        url: relativePath,
+        hostname: `${cidv1Address ?? address}.${namespace}.${request.hostname}`
+      },
+      reply,
+      overrideResponseCode: 301
+    })
```

Comment:
upon further investigation, I don't think this is what kubo is doing. I'm adding a test so we can mimic functionality. see attached.